### PR TITLE
Add bump threshold

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Add a threshold for bumping entries.

### CODE REVIEW

File: soroban-env-host/src/storage.rs [hunk 1]
```diff
@@ -332,7 +333,7 @@ impl Storage {
             )
         })?;
 
-        if new_expiration > old_expiration {
+        if new_expiration > old_expiration && new_expiration - old_expiration >= threshold {
```
Comment (dmkozh) [inline soroban-env-host/src/storage.rs:336]: We need to fail if it's never possible to bump the entry, i.e. if `bump_by_ledgers <= threshold`.

Comment (sisuresh) [reply soroban-env-host/src/storage.rs:336]: The name of `bump_by_ledgers` probably needs to change - it's not the actual bump amount. `bump_by_ledgers` is the bump from the current ledger, but doesn't account for the difference between old_expiration and current ledger. This means you could have a bump_by_ledgers of 10, but new_expiration-old_expiration could be greater than 10.

This brings up a question though - should the threshold be on the bump from the old_expiration ledger to new_expiration, or from old_expiration to current ledger?

Comment (sisuresh) [reply soroban-env-host/src/storage.rs:336]: A better way to phrase this - should the threshold be on a minimum bump, or a threshold on how close you have to be to expiration before you can bump again.

Comment (sisuresh) [reply soroban-env-host/src/storage.rs:336]: Ah yeah my bad...

Comment (sisuresh) [reply soroban-env-host/src/storage.rs:336]: Shouldn't the check be `bump_by_ledgers < threshold`? If current ledger == 10, expiration == 10, bump_by_ledgers == 2, and threshold == 2, the new expiration would be 12 with a bump of 2 right?

**Pull request closed without merging**